### ISSUE TITLE
Ensure external spdlog library can be used

### ISSUE THREAD

**Issue by nightlark** [GMLC-TDC/HELICS#2297]
A user was having issues with the HELICS copy of spdlog interfering with other copies of spdlog. Make sure there is a way in HELICS to use an external/system copy of spdlog. Specifically, this needs to work when using spack.

**Comment by phlptp** [comment]
@nightlark  do you have a PR for this close to ready?

**Comment by nightlark** [comment]
Yes, just need to test all the combinations of external spdlog+fmt.
